Competition/src/main/cpp/Arm.cpp

```diff
@@ -0,0 +1,98 @@
+#include "Arm.h"
+
+Arm::Arm() : armMtrLeft{ArmConstants::TALON_ID_LEFT_ARM}, 
+             armMtrRight{ArmConstants::TALON_ID_RIGHT_ARM} {
+
+}
+
+Arm::Arm(frc::XboxController* controller) : operator_controller(controller), 
+                                            armMtrLeft{ArmConstants::TALON_ID_LEFT_ARM}, 
+                                            armMtrRight{ArmConstants::TALON_ID_RIGHT_ARM} {
+    InitArm();
+}
+
+Arm& Arm::GetInstance() {
+    static Arm instance;
+    return instance;
+}
+
+void Arm::Periodic() {
+    assessInputs();
+    assignOutputs();
+}
+
+void Arm::InitArm() {
+    armMtrLeft.ConfigFactoryDefault();
+    armMtrRight.ConfigFactoryDefault();
+
+    armMtrLeft.SetNeutralMode(NeutralMode::Brake);
+    armMtrRight.SetNeutralMode(NeutralMode::Brake);
+
+    armMtrLeft.SetInverted(false);
+    armMtrRight.SetInverted(false);
+
+    state.disengage = true;
+    state.step1_start_time = -1;
+    state.step2_start_time = -1;
+}
+
+void Arm::setDefaultState() {
+    state.arm_state = ArmState::DISABLED;
+}
+
+void Arm::assessInputs() {
+    if (state.disengage) {
```

Comment:
One of the artificial rules we set was that you could not read the state variable from within the assessInputs function. This is to make sure ALL decisions are made within the assignOutputs function.

Therefore this if statement should be moved into assignOutputs. The else case is perfect though

Reply:
Just confirming, so having this if statement just to set state.arm_state in AssignOutputs() is fine then? It would check the boolean for disengage and assign the state if true and then jump into the other AssignOutputs() logic after.

Reply:
Yup that works!